client/src/pages/spectrogram/components/PluginsPane.tsx

```diff
@@ -94,7 +98,7 @@ export const PluginsPane = ({ cursorsEnabled, handleProcessTime, meta, setMeta }
           toast.error(`Plugin failed to run: ${data.status}`);
           return;
         }
-        if (data.data_output && data.data_output.length > 0) {
+        if (data.data_output[0]['data_type'] == IQ_SIGMF_32 && data.data_output && data.data_output.length > 0) {
```

Comment:
would it be better to do a find on data_type here which would return the first element with that data type?
const dataType = data.data_output.find(obj => obj['data_type'] === 'IQ_SIGMF_32');

Reply:
Ordinarily I would agree with you. But there's a comment in the code saying that 99% of plugins have only one return object, so handling of higher elements is disregarded throughout.